xbmc/addons/AddonRepos.cpp

```diff
@@ -33,6 +32,22 @@ static std::vector<RepoInfo> officialRepoInfos = CCompileInfo::LoadOfficialRepoI
  *
  */
 
+CAddonRepos::CAddonRepos() : m_addonMgr(CServiceBroker::GetAddonMgr())
+{
+  m_valid = m_addonDb.Open() && LoadAddonsFromDatabase("", nullptr);
+}
+
+CAddonRepos::CAddonRepos(const std::string& addonId) : m_addonMgr(CServiceBroker::GetAddonMgr())
+{
+  m_valid = m_addonDb.Open() && LoadAddonsFromDatabase(addonId, nullptr);
+}
+
+CAddonRepos::CAddonRepos(const std::shared_ptr<IAddon>& repoAddon)
+  : m_addonMgr(CServiceBroker::GetAddonMgr())
+{
+  m_valid = m_addonDb.Open() && LoadAddonsFromDatabase("", repoAddon);
```

Comment:
What is the meaning of `""` here? Would a string constant be better?

Reply:
like this?
```
+namespace
+{
+constexpr auto ALL_ADDON_IDS = "";
+constexpr auto ALL_REPOSITORIES = nullptr;
+} // anonymous namespace
+
 using namespace ADDON;
 
 static std::vector<RepoInfo> officialRepoInfos = CCompileInfo::LoadOfficialRepoInfos();
@@ -35,19 +41,19 @@ static std::vector<RepoInfo> officialRepoInfos = CCompileInfo::LoadOfficialRepoI
 CAddonRepos::CAddonRepos()
   : m_addonMgr(CServiceBroker::GetAddonMgr()), m_addonDb(CServiceBroker::GetAddonDatabase())
 {
-  m_valid = m_addonDb.IsOpen() && LoadAddonsFromDatabase("", nullptr);
+  m_valid = m_addonDb.IsOpen() && LoadAddonsFromDatabase(ALL_ADDON_IDS, ALL_REPOSITORIES);
 }
 
 CAddonRepos::CAddonRepos(const std::string& addonId)
   : m_addonMgr(CServiceBroker::GetAddonMgr()), m_addonDb(CServiceBroker::GetAddonDatabase())
 {
-  m_valid = m_addonDb.IsOpen() && LoadAddonsFromDatabase(addonId, nullptr);
+  m_valid = m_addonDb.IsOpen() && LoadAddonsFromDatabase(addonId, ALL_REPOSITORIES);
 }
 
 CAddonRepos::CAddonRepos(const std::shared_ptr<IAddon>& repoAddon)
   : m_addonMgr(CServiceBroker::GetAddonMgr()), m_addonDb(CServiceBroker::GetAddonDatabase())
 {
-  m_valid = m_addonDb.IsOpen() && LoadAddonsFromDatabase("", repoAddon);
+  m_valid = m_addonDb.IsOpen() && LoadAddonsFromDatabase(ALL_ADDON_IDS, repoAddon);
 }
```